packages/data-store/src/utils/presentationDefinition/MappingUtils.ts

```diff
@@ -0,0 +1,56 @@
+import { PresentationDefinitionItemEntity } from '../../entities/presentationDefinition/PresentationDefinitionItemEntity'
+import { IPresentationDefinition } from '@sphereon/pex'
+import { PartialPresentationDefinitionItem, PresentationDefinitionItem } from '../../types'
+import { createHash } from 'crypto'
+
+export const presentationDefinitionItemFrom = (entity: PresentationDefinitionItemEntity) => {
+  const item: PresentationDefinitionItem = {
+    id: entity.id,
+    tenantId: entity.tenantId,
+    definitionId: entity.definitionId,
+    version: entity.version,
+    purpose: entity.purpose,
+    definitionPayload: JSON.parse(entity.definitionPayload) as IPresentationDefinition,
+    createdAt: entity.createdAt,
+    lastUpdatedAt: entity.lastUpdatedAt,
+  }
+  return item
+}
+
+export const presentationDefinitionEntityItemFrom = (item: PartialPresentationDefinitionItem) => {
+  const entity = new PresentationDefinitionItemEntity()
+  if (item.id) {
+    entity.id = item.id
+  }
+
+  entity.tenantId = item.tenantId
+  entity.definitionId = item.definitionId!
+  entity.version = item.version!
+  entity.purpose = item.purpose
+  entity.definitionPayload = JSON.stringify(item.definitionPayload!)
+  entity.createdAt = item.createdAt!
+  entity.lastUpdatedAt = item.lastUpdatedAt!
+  return entity
+}
+
+function hashPayload(payload: IPresentationDefinition): string {
+  return createHash('md5').update(JSON.stringify(payload)).digest('hex')
+}
+
+export function isPresentationDefinitionEqual(left: PartialPresentationDefinitionItem, right: PartialPresentationDefinitionItem): boolean {
+  if (
+    left.definitionId !== right.definitionId ||
+    left.tenantId != right.tenantId ||
+    left.version !== right.version ||
+    left.purpose != right.purpose
+  ) {
+    return false
+  }
+
+  if (left.definitionPayload && right.definitionPayload) {
+    return hashPayload(left.definitionPayload) === hashPayload(right.definitionPayload)
+  }
+
+  // return false when either or both are null or undefined
```

Comment:
validations should run before the business logic execution.
so if we want to return `false` whenever one or both `definitionPayload` are `null` or `undefined`, we return that right away and guarantee that from that point on we know both objects have something in `definitionPayload`
then all we need to do is check whether they payload is the same by returning the current line 51 `return hashPayload(left.definitionPayload) === hashPayload(right.definitionPayload)`

Reply:
@jcmelati 

That would become something more verbose
```
export function isPresentationDefinitionEqual(left: PartialPresentationDefinitionItem, right: PartialPresentationDefinitionItem): boolean {
  if (
    left.definitionId !== right.definitionId ||
    left.tenantId !== right.tenantId ||
    left.version !== right.version ||
    left.purpose !== right.purpose
  ) {
    return false;
  }

  if (left.definitionPayload === undefined && right.definitionPayload === undefined) {
    return true;
  }

  if ((left.definitionPayload === undefined || right.definitionPayload === undefined) && left.definitionPayload !== right.definitionPayload) {
    return false;
  }

  return hashPayload(left.definitionPayload) === hashPayload(right.definitionPayload);
}
```
Is that preferred?

Reply:
> validations should run before the logic execution
Technically it still does that in the original scenario. 
if (base.definitionPayload && compare.definitionPayload)  is validation and hashPayload won't execute until that passed

Reply:
@jcmelati 
It's now like this
```

export function isPresentationDefinitionEqual(base: PartialPresentationDefinitionItem, compare: PartialPresentationDefinitionItem): boolean {
  if (
    base.definitionId !== compare.definitionId ||
    base.tenantId != compare.tenantId ||
    base.version !== compare.version ||
    base.name != compare.name ||
    base.purpose != compare.purpose
  ) {
    return false
  }

  if (base.definitionPayload && compare.definitionPayload) {
    return hashPayload(base.definitionPayload) === hashPayload(compare.definitionPayload)
  }

  return false
}
```

ok?

Reply:
Nah still don't like the approach, but don't want us to get stuck on this